### PR TITLE
16158 fix multiple select validation

### PR DESCRIPTION
**Pull request by np25071984**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #16158

### CODE REVIEW

**Comment by GHopperMSK** [comment]
I am not sure we should follow current recommendation of codeclimate. It will increase the methods complexity. What do you think?

**Comment by samdark** [comment]
I think in this case it is better to ignore it.

File: framework/assets/yii.activeForm.js [hunk 1]
```diff
@@ -565,6 +565,70 @@
...
+        if (Array.isArray(val1)) {
+            return isArraysEqual(val1, val2);
+        }
+
+        // simple types
+        return val1 === val2;
+    };
+
+    var isObjectsEqual = function(obj1, obj2) {
+        if (!(obj1 instanceof Object)) {
```
Comment (samdark) [inline framework/assets/yii.activeForm.js]:
```suggestion
        if (!(obj1 instanceof Object) || !(obj2 instanceof Object)) {
```

File: framework/assets/yii.activeForm.js [hunk 2]
```diff
@@ -565,6 +565,70 @@
...
+            if (obj1[keys1[i]] !== obj2[keys1[i]]) {
+                return false;
+            }
+        }
+
+        return true;
+    };
+
+    var isArraysEqual = function(arr1, arr2) {
+        if (!Array.isArray(arr1)) {
```
Comment (samdark) [inline framework/assets/yii.activeForm.js]:
```suggestion
        if (!Array.isArray(arr1) || !Array.isArray(arr2)) {
```

File: framework/assets/yii.activeForm.js [hunk 3]
```diff
@@ -565,6 +565,70 @@
...
+     * @returns boolean
+     */
+    var isEqual = function(val1, val2) {
+        // objects
+        if (val1 instanceof Object) {
+            return isObjectsEqual(val1, val2)
+        }
+
+        // arrays
+        if (Array.isArray(val1)) {
```
Comment (samdark) [inline framework/assets/yii.activeForm.js]: Arrays are objects as well. Is special case necessary?

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/assets/yii.activeForm.js (hunk 3 above)
Comment (GHopperMSK) [reply framework/assets/yii.activeForm.js]: Yes, we do need the special case due to different approach to compare them.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**